### ISSUE TITLE
possible display issue when using code blocks mixed with normal text in multiline

### ISSUE THREAD

**Issue by nipsy** [wee-slack/wee-slack#880]
This might not be a change in behavior, but I thought this worked differently in the past.

I will often type code blocks mixed in with my normal text, and I thought in the past, that I could simply do something like:

```
Hey everyone, check this:
```ls
echo foo bar```
wow
```

which currently renders like this graphically in Slack:

```
Hey everyone, check this:
ls
echo foo bar
```
wow

---

If I make sure to add an extra blank line between the first line and the start of the code block like this:

```
Hey everyone, check this:

```ls
echo foo bar```
wow
```

then everything renders correctly in Slack (app or browser) like this:

Hey everyone, check this:

```
ls
echo foo bar
```

wow

---

Everything comes through literally as I've typed it of course in wee-slack, so I didn't notice until somewhat recently I was sending muddled code blocks to folks, since I rarely look at official clients.  Also note that the bottom line doesn't require the extra blank newline and renders as expected outside of the code block.

I think this worked in the expected fashion in the past where an extra blank newline wasn't required between the first line and the start of the code block, but I'm possibly mistaken.  Either way, I figured it was worth opening an issue about to double check I'm not losing my mind.  I know things changed a fair amount recently to account for the Slack API deprecating some stuff, so maybe this was a byproduct of that somehow.

Anyway, the workaround is good enough, but hopefully this behavior can be restored/implemented.  Thanks.

**Comment by trygveaa** [comment]
Is this still happening for you? I tried now, and it renders correctly for me in the web client no matter how I send it.

I also made [a change](https://github.com/wee-slack/wee-slack/commit/74da30342e8d328fc085d2158e37afda3b908f2c) now on how messages are rendered in wee-slack which uses the structured message to render it, so everything that is in code blocks in the web client should be inside ```` ```...``` ```` in wee-slack as well.

**Comment by nipsy** [comment]
Yes and no, although now I'm assuming the bug is actually on the Slack side.  It looked okay through the web interface when I just tested everything with Weechat 4.0.1, 0.6.4 of multiline.pl loaded and wee-slack at the latest commit of 2ee278623b2f7a1fac8a6908999cb8e33a1be13f.  However, it's still combining everything into the code block without an empty newline as a delimiter between a line and a code block on the current version of the official Slack Android app!

So I guess we can probably call this done from this end, unless you have some deeper insight as to how this might still be a wee-slack issue.

Thanks!

**Comment by trygveaa** [comment]
Ah, I didn't test with the Android app. I checked now and see the same as you. That's very strange though, as what's shown in the app doesn't at all match what I'm getting from the API.

In order to fix this we probably have to send a structured message when posting, rather than just a string. E.g. split the message into normal text and code blocks and post it to slack with information on what is what (this is how the web client manages to create a message that doesn't have a blank line between the text and the code block and which still looks correct in the Android app).

Anther alternative is to create a bug report with Slack.

**Comment by nipsy** [comment]
I've tried submitting an internal help request via my work account since I couldn't seem to find a better place to submit bugs to them.  I guess we'll see how that goes.

**Comment by nipsy** [comment]
I was told via their Zendesk support folks that this is a known bug on their end and that my ticket was attached to the bug report so I'd be automatically notified as soon as a fix becomes available.  Sadly, no actual tracking information was provided in any of their response, so we'll just have to accept all of that on faith.

Feel free to do what you want with this particular issue as I think we've exhausted all the options at this point.  I'm content with knowing I can still keep the newline delimiter and things will mostly look as I intend them, regardless of the client being used.
